direct/nn/openvino/openvino_model.py

```diff
@@ -0,0 +1,107 @@
+    """
+    This is a support class which helps export network with InstanceNorm2d in ONNX format.
+    """
+
+    def __init__(self, num_features):
+        super().__init__(num_features)
+        self.origin_forward = super().forward
+        self.scale_one = torch.ones(num_features)
+        self.bias_zero = torch.zeros(num_features)
+
+    def forward(self, input):
+        return InstanceNorm2dFunc.apply(self, input).clone()
+
+
+def convert_layer(model):
+    for name, l in model.named_children():
+        layer_type = l.__class__.__name__
+        if layer_type == "InstanceNorm2d":
+            new_layer = InstanceNorm2dONNX(l.num_features)
+            setattr(model, name, new_layer)
+        else:
+            convert_layer(l)
+
+
+class OpenVINOModel(nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.inputs: tuple
+        self.input_names: list
+        self.model = model
+        self.exec_net = None
+        self.model_name = self.model.__class__.__name__
+
+    def create_net(self, *args, **kwargs):
+        self.inputs = args
+
+        if self.model_name == "RIM":
+            self.input_names = ["input_image", "masked_kspace", "sampling_mask", "sensitivity_map"]
+            output_names = ["cell_outputs", "previous_state"]
+        elif self.model_name == "Unet2d":
+            output_names = ["output"]
+            self.input_names = ["masked_kspace"]
+            if self.model.image_initialization == "sense":
+                self.input_names.append("sensitivity_map")
+        else:
+            raise ValueError(f"The model is not supported by OpenVINO: {self.model_name}")
+
+        for k in self.input_names:
+            if k in kwargs:
+                self.inputs += tuple([kwargs[k]])
+
+        ie = IECore()
+        ie.add_extension(get_extensions_path(), "CPU")
+
+        convert_layer(self.model)
+
+        with torch.no_grad():
+            buf = io.BytesIO()
+            torch.onnx.export(
+                self.model,
+                self.inputs,
+                buf,
+                opset_version=12,
+                operator_export_type=torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH,
+                input_names=self.input_names,
+                output_names=output_names,
+            )
+
+            net = ie.read_network(buf.getvalue(), b"", init_from_buffer=True)
+            self.exec_net = ie.load_network(net, "CPU")
+
+    def postprocess(self, res):
+        if self.model_name == "RIM":
+            out = ([torch.Tensor(res["cell_outputs"])], torch.Tensor(res["previous_state"]))
+        elif self.model_name == "Unet2d":
+            out = torch.Tensor(res["output"])
+        return out
+
+    def forward(self, *args, **kwargs):
+        if self.exec_net is None:
+            self.create_net(*args, **kwargs)
+
+        ov_input = dict(zip(self.input_names, self.inputs))
+        res = self.exec_net.infer(ov_input)
+
+        return self.postprocess(res)
```

Comment:
Please make sure that all new methods/classes have docstrings and are adequately documented.

Reply:
Done